.github/scripts/check_lineage_updates.sh

```diff
@@ -1,176 +1,32 @@
-old_changed_plugin_bad_info = straxen.test_utils.bad_field_info(st, run_id, list(new_changed_plugin_bad_info.keys()))
-old_affected_plugin_bad_info = straxen.test_utils.bad_field_info(st, run_id, list(new_affected_plugin_bad_info.keys()))
-
-###Now report the differences
-#Lowest level plugins
-all_plugin_change_info = {"Lowest Levels":{'old':old_changed_plugin_bad_info,
-    'new':new_changed_plugin_bad_info},
-    "Affected":{'old':old_affected_plugin_bad_info,
-    'new':new_affected_plugin_bad_info}}
-
-for level in ['Lowest Levels', 'Affected']:
-    print(f"#################### {level} Plugins ####################")
-
-    for p in all_plugin_change_info[level]['old']:
-        print(f"Change report for '{p}':")
-        data_types_old = np.array(list(all_plugin_change_info[level]['old'][p].keys()))[::2]
-        data_types_new = np.array(list(all_plugin_change_info[level]['new'][p].keys()))[::2]
-
-        all_data_types = np.unique(np.concatenate([data_types_old, data_types_new]))
-        for d in all_data_types:
-            if d not in data_types_old:
-                print(f"    - New column {d} added")
-            elif d not in data_types_new:
-                print(f"    - Column {d} deleted")
-            else:
-                if (all_plugin_change_info[level]['old'][p][d] != all_plugin_change_info[level]['new'][p][d]):
-                    print(f"    - {d} bad fraction changed from: {all_plugin_change_info[level]['old'][p][d]} -> {all_plugin_change_info[level]['new'][p][d]}")
-                if (all_plugin_change_info[level]['old'][p][f'mean_{d}'] != all_plugin_change_info[level]['new'][p][f'mean_{d}']):
-                    print(f"    - {d} mean value changed from: {all_plugin_change_info[level]['old'][p][f'mean_{d}']} -> {all_plugin_change_info[level]['new'][p][f'mean_{d}']}")
-        print("All other columns remained the same\n")
-    print('\n')
-EOF
-)
-
-python3 -c "$compare_plugins" "$@"
-
-#Remove the temporary dictionaries that were created to deal with switching branches
-rm new_affected_plugin_bad_info.json
-rm new_changed_plugin_bad_info.json
-rm plugin_update_comparison.json
-rm version_hash_dict_new.json
-rm version_hash_dict_old.json
+top_level=$(git rev-parse --show-toplevel)
+current_branch=$(git rev-parse --abbrev-ref HEAD)
+
+# run cleanup on exit
+function cleanup {
+    git checkout $current_branch
+}
+trap cleanup EXIT
+
+cd $top_level/bin
+
+git checkout $old_branch
+./report_pr_changes --branch old --computation lineage_hash
+git checkout $new_branch
+./report_pr_changes --branch new --computation lineage_hash
+./report_pr_changes --branch new --computation hash_comparison
+./report_pr_changes --branch new --computation print_added_plugin
+./report_pr_changes --branch new --computation changed_affected_plugin
+git checkout $old_branch
+./report_pr_changes --branch old --computation changed_affected_plugin
+./report_pr_changes --branch old --computation report_changes
```

Comment:
Does the cleanup function automatically delete all of the temporary json files that we've made?

Reply:
No, I have not deleted them. Maybe should delete it by python scripts. Because they are generated by the python scripts.